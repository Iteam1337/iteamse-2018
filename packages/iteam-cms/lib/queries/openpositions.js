module.exports = async (_, {}, { getByContentType }) => {
  try {
    const content = await getByContentType('annons')

    return content.map(item =>
      Object.assign({}, item.fields, {
        id: item.fields.urlId,
        headerImage: item.fields.headerImage.fields.file.url,
        contentImage:
          item.fields.contentImage && item.fields.contentImage.fields.file.url,
      })
    )
  } catch (error) {
    throw new Error(error)
  }
}
