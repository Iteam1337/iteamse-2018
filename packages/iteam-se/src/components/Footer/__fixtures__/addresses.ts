export const addresses = [
  {
    __typename: 'Address',
    address1: 'Haga Kyrkogata 28',
    city: 'Göteborg',
    contactMail: 'johanna.grahn@iteam.se',
    contactPhone: '+46 734 029 112',
    orgNumber: '556551-6928',
    title: 'Iteam Göteborg',
    zip: '411 23',
  },
  {
    __typename: 'Address',
    address1: 'Östermalmsgatan 26A',
    city: 'Stockholm',
    contactMail: 'info@iteam.se',
    contactPhone: '+46 826 70 90',
    orgNumber: '556551-6928',
    title: 'Iteam Stockholm',
    zip: '114 26',
  },
]
