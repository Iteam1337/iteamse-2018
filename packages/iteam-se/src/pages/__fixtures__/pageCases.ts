export const pageCases = {
  __typename: 'PageCases',
  contactTitle: 'Hör av dig till oss!',
  headerImage:
    '//images.ctfassets.net/rj4r6yfcesw5/45ST2zZeFqgYuaSuM8yKy0/455471362d58784653089b0f77b2c4e9/Maria-Desk-2-Highres.jpg',
  headerText1: 'Några exempel på',
  headerText2: 'sådant vi gjort',
  headerTextBgColor: 'green',
  team: ['rln', 'msr'],
}
