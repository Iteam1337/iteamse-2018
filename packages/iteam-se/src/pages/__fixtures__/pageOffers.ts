// tslint:disable:object-literal-sort-keys
export const pageOffers = {
  headerImage:
    '//images.ctfassets.net/rj4r6yfcesw5/45ST2zZeFqgYuaSuM8yKy0/d7d62b50748a05f1f9f540198390077a/Maria-Desk-2-Highres.jpg',
  headerText1: 'Våra erbjudanden',
  headerText2: null,
  headerTextBgColor: 'green',
  offersLeadText:
  'Kod, kultur och strategi är nödvändliga beståndsdelar för att skapa innovation. Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus.',
  contactTitle: 'Vill du veta mer?',
  team: ['mrs', 'jmn'],
  __typename: 'PageOffers',
}
