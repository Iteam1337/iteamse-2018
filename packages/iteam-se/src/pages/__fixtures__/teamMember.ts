export const teamMember = {
  __typename: 'TeamMember',
  avatar:
    'https://www.gravatar.com/avatar/08053b9b0035a1ad64dc9be5485f2cf5?s=300',
  background:
    'Materials Science and Engineering at KTH (Royal Institute of Technology). Interaction designer at Nackademin.',
  backgroundTitle: 'Bakgrund',
  competence: 'HTML5, CSS and JavaScript.',
  competenceTitle: 'Specialistkompetens',
  email: 'rickard.laurin@iteam.se',
  headerImage:
    '//images.ctfassets.net/rj4r6yfcesw5/2ogLKEElzmaIUmQsU0UYCi/15ff16e00c9edd8f562fb483e7784886/35828573134_4c9fbe38d4_k.jpg',
  headerTextBgColor: 'blue',
  location: 'Stockholm',
  name: 'Rickard Laurin',
  phoneNumber: null,
  team: ['rln', 'msr'],
  title: 'Webbutvecklare',
  why:
    'In the early 90\'s my family got its first 286, it was love at first sight. In my teen years I played a lot of Delta Force and started developing the clan websites. Until I started at KTH I only did minor hobby projects. There I managed my chapters website and assisted younger students in MATLAB. This is when I realized what I really wanted to work with, building beautiful websites.',
  whyTitle: 'Varf√∂r jobbar du med teknik?',
}
