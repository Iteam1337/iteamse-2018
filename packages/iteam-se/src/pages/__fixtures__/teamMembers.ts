export const teamMembers = [
  {
    __typename: 'TeamMember',
    avatar: '',
    email: 'cm@sesame.com',
    gravatar: 'cm.png',
    location: 'Stockholm',
    name: 'Cookie M',
    phoneNumber: '0701234567',
    short: 'cmm',
    title: 'Cookie master',
  },
  {
    __typename: 'TeamMember',
    avatar: '',
    email: 'count@sesame.com',
    gravatar: 'count.png',
    location: 'GÃ¶teborg',
    name: 'Count van Count',
    phoneNumber: '0707654321',
    short: 'cvc',
    title: '1 ahahah',
  },
]
