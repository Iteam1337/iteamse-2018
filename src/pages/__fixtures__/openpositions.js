export const openpositions = [
  {
    id: 'junior-ux-designer',
    location: 'Stockholm',
    role:
      'Som UX designer på Iteam är din roll att hjälpa våra kunder med digitala lösningar som skapar värde på riktigt. Som gör något bra både för kunden, slutanvändaren och samhället. Med hjälp av olika researchmetoder undersöker du både användarens och kundens behov och kontext, samt definierar och testar användarupplevelser.\n\nDu delar med dig av din kunskap inom user experience. Du förstår också att om lösningar ska vara verkligt användarvänliga behöver även arbetsprocesserna och kulturen vara användarcentrerad. Vi strävar mot att UX aldrig ska få vara bara en fas eller en eftertanke. Det ska genomsyra hela projektet. Alltid. ',
    title: 'Junior UX Designer',
    __typename: 'OpenPosition',
  },
]
