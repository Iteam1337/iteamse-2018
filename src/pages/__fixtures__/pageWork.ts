export const pageWork = {
  __typename: 'PageWork',
  headerImage:
    '//images.ctfassets.net/rj4r6yfcesw5/j9IXvqaJ32Uc0SWQeUQsi/e75bc2d9b05ae39fe71ec1b1c14600e0/work.jpg',
  headerText1: 'Bli en del av teamet',
  headerText2: null,
  headerTextBgColor: '#FF405F',
}
