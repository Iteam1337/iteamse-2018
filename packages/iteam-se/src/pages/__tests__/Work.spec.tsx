import React from 'react'
import { render, wait } from 'react-testing-library'
import MockedQuery from '../../utils/test-utils/MockedQuery'
import { openpositions } from '../__fixtures__/openpositions'
import { pageWork } from '../__fixtures__/pageWork'
import { TeamQueryMock } from '../__fixtures__/teamMock'
import Work, { WORK_PAGE_QUERY } from '../Work'

const mocks = [
  {
    request: {
      query: WORK_PAGE_QUERY,
    },
    result: {
      data: {
        openpositions,
        pageWork,
      },
    },
  },
  TeamQueryMock,
]

describe('components/Work', () => {
  it('renders Work', async () => {
    const { container } = render(
      <MockedQuery mocks={mocks}>
        <Work />
      </MockedQuery>
    )

    await wait()

    expect(container).toMatchSnapshot()
  })
})
