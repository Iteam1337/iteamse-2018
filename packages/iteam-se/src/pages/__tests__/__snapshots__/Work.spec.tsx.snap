// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Work renders Work 1`] = `
<div>
  <div>
    <div
      class="GridColumn__GridColumnClean-sc-1y5e0ej-0 sc-kkGfuU jdkWvC"
    >
      <div
        class="Header__Content-dm3ycw-0 beGtUJ"
      >
        <div
          class="Navigation__Wrap-fdgx9g-0 ehbMFf"
        >
          <a
            class="sc-jKJlTe bIUarX"
          >
            <img
              data-testid="logo"
              path="test-file-stub"
              class="sc-eNQAEJ gYmRno"
            />
          </a>
          <div
            class="Navigation__NavigationItems-fdgx9g-2 eVDHVt"
          >
            <a
              class="sc-hMqMXs fILlfv"
              aria-current="false"
            >
              Erbjudanden
            </a>
            <a
              class="sc-hMqMXs fILlfv"
              aria-current="false"
            >
              Våra case
            </a>
            <a
              class="sc-hMqMXs fILlfv"
              aria-current="false"
            >
              Metod
            </a>
            <a
              class="sc-hMqMXs fILlfv"
              aria-current="false"
            >
              Medarbetare
            </a>
            <a
              class="sc-hMqMXs fILlfv"
              aria-current="false"
            >
              Karriär
            </a>
            <a
              class="sc-hMqMXs fILlfv"
              aria-current="false"
            >
              Om
            </a>
            <a
              target="_blank"
              rel="noopener noreferrer"
              href="https://blog.iteam.life/"
              class="NativeLink-fvundc-0 sc-kEYyzF grsRKu"
            >
              Labs
            </a>
            <div
              class="Navigation__Indicator-fdgx9g-1 bOcqdk"
            />
          </div>
        </div>
        <div
          class="Header__MessageRow-dm3ycw-1 cbRVND"
        >
          <h1
            class="H1-sc-1we7oaz-0 sc-iAyFgw cylnsh"
          >
            <span
              class="Header__MessageLine-dm3ycw-2 iOLoPT"
            >
              Bli en del av teamet
            </span>
          </h1>
        </div>
      </div>
    </div>
  </div>
  <div
    class="GridColumn__GridColumnClean-sc-1y5e0ej-0 sc-htpNat kmHAwx"
  >
    <div
      class="react-reveal"
      style="opacity: 1;"
    >
      <div
        data-test="block-location"
        class="PaddedRow-qwtz4d-0 glcQGC"
      >
        <div
          class="Block__Content-sc-73lkvi-0 cmDqNt"
        >
          <div
            class="Block__TitleWrap-sc-73lkvi-1 jjTrxf"
          >
            <a
              class="sc-bdVaJa bssnas"
            >
              <h3
                class="H3-sc-13x7erc-0 cKRnEQ"
              >
                Junior UX Designer
              </h3>
            </a>
            <div
              class="Block__Subtitle-sc-73lkvi-2 hqFiHL"
            >
              Stockholm
            </div>
          </div>
          <div
            class="Block__Text-sc-73lkvi-3 jzbETd"
          >
            <div
              class="Paragraph__StyledParagraph-sc-18re3wf-0 kFBGJT"
            >
              <div>
                <p>
                  Som UX designer på Iteam är din roll att hjälpa våra kunder med digitala lösningar som skapar värde på riktigt. Som gör något bra både för kunden, slutanvändaren och samhället. Med hjälp av olika researchmetoder undersöker du både användarens och kundens behov och kontext, samt definierar och testar användarupplevelser.
                </p>
                <p>
                  Du delar med dig av din kunskap inom user experience. Du förstår också att om lösningar ska vara verkligt användarvänliga behöver även arbetsprocesserna och kulturen vara användarcentrerad. Vi strävar mot att UX aldrig ska få vara bara en fas eller en eftertanke. Det ska genomsyra hela projektet. Alltid. 
                </p>
              </div>
            </div>
            <a
              black="0"
              class="sc-gZMcBi foZFVe"
            >
              Läs mer och ansök
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="react-reveal"
      style="opacity: 1;"
    >
      <div
        data-test="block-"
        class="PaddedRow-qwtz4d-0 glcQGC"
      >
        <div
          class="Block__Content-sc-73lkvi-0 cmDqNt"
        >
          <div
            class="Block__TitleWrap-sc-73lkvi-1 jjTrxf"
          >
            <h3
              class="H3-sc-13x7erc-0 cKRnEQ"
            >
              Inget som passar?
            </h3>
          </div>
          <div
            class="Block__Text-sc-73lkvi-3 jzbETd"
          >
            <div
              class="Paragraph__StyledParagraph-sc-18re3wf-0 kFBGJT"
            >
              <div>
                <p>
                  Gör en 
                  <a
                    href="mailto:joinus@iteam.se"
                  >
                    spontanansökan
                  </a>
                  !
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
